test/distributed/test_data_parallel.py

```diff
@@ -132,8 +132,8 @@ def forward(self, var1, var2, float1, var3=None):
         gvar2_exp = var2.grad.clone()
 
         def local_test(out):
-            var1.grad.data.fill_(0.0)
-            var2.grad.data.fill_(0.0)
+            var1.grad.fill_(0.0)
```

Comment:
Is it fine to do this update in a differentiable manner? Or do we want a `with torch.no_grad():` ? 
Same below?